### PR TITLE
Add JWT Support for Authentication

### PR DESCRIPTION
**Pull request by baixinsui**
fixed https://github.com/eclipse-xpanse/xpanse/issues/793
related https://github.com/eclipse-xpanse/xpanse-iam/pull/9

### CODE REVIEW

**Review by swaroopar — CHANGES_REQUESTED** [review]
@baixinsui actual implementation looks great. only question about the terraform boot client class change.

File: modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/deployers/terraform/terraformboot/ApiClient.java [hunk 1]
```diff
@@ -1,34 +1,5 @@
 package org.eclipse.xpanse.modules.deployment.deployers.terraform.terraformboot;
 
-import org.apache.commons.logging.Log;
```
Comment (iskey) [inline modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/deployers/terraform/terraformboot/ApiClient.java:3]: Can we remove this change from this PR?

File: modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/deployers/terraform/terraformboot/ApiClient.java [hunk 2]
```diff
@@ -51,34 +22,54 @@
 import java.util.Map;
 import java.util.Map.Entry;
 import java.util.TimeZone;
-import java.time.OffsetDateTime;
-
+import org.apache.commons.logging.Log;
```
[the remark this answers is not in the document]
Comment (baixinsui) [reply modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/deployers/terraform/terraformboot/ApiClient.java:25]: Yes. I will restore it in this PR.

File: modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/deployers/terraform/terraformboot/ApiClient.java (hunk 1 above)
Comment (baixinsui) [reply modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/deployers/terraform/terraformboot/ApiClient.java]: Ok.

**Review by swaroopar — APPROVED** [review]
LGTM

**Comment by swaroopar** [comment]
@baixinsui looks good to me. We can merge if Yichao has no further comments

**Review by iskey — APPROVED** [review]
looks good

**Pull request merged**